script.js

```diff
@@ -1,82 +1,217 @@
+  filterSelectElement.className = "form-control p-1 m-1";
+  parentOfSelectElement.append(filterSelectElement);
+  filterDivElement.append(
+    filteringLabelElement,
+    parentOfInputElement,
+    filterFoundLabelElement,
+    parentOfSelectElement
   );
+  filterSectionElement.append(filterDivElement);
+  const rootElement = document.getElementById("root");
+  rootElement.append(filterSectionElement);
+  fillShowsDropDown();
+  filterSelectElement.addEventListener("change", function () {
+    const rootElem = document.getElementById("root");
+    const showSectionElement = document.getElementById("displayShows");
+    rootElem.removeChild(showSectionElement);
+    let selectedShow = getSelectedShow(this.value);
+    makeShowSection(selectedShow);
+  });
+
+  function getSelectedShow(selectedId) {
+    return getAllShows().filter((show) => show.id == selectedId);
+  }
+  filterInputElement.addEventListener("input", function () {
+    const rootElem = document.getElementById("root");
+    const showSectionElement = document.getElementById("displayShows");
+    rootElem.removeChild(showSectionElement);
+    let filteredShows = fillShowsDropDown();
+    makeShowSection(filteredShows);
+  });
 }
 
-function fillSelectByShows(selectedShow) {
-  let searchSection = document.getElementById("searchSection");
-  let SelectElement = document.getElementById("showSelect");
-  let allShows = getAllShows();
+// Display show/shows
+function makeShowSection(displayedShows) {
+  const rootElem = document.getElementById("root");
+  let showSectionElement = document.createElement("section");
+  showSectionElement.className = "p-2";
+  showSectionElement.id = "displayShows";
+  showSectionElement.style.display = "block";
+  rootElem.appendChild(showSectionElement);
+  let allShows =
+    displayedShows == null || displayedShows.length == 0
+      ? getAllShows()
+      : displayedShows;
+  allShows.sort((a, b) => (a.name > b.name ? 1 : -1));
   allShows.map((show) => {
-    let optionText = show.name;
-    let optionElement = document.createElement("option");
-    optionElement.innerText = show.name;
-    optionElement.value = show.id;
-    SelectElement.appendChild(optionElement);
+    let ShowElement = document.createElement("div");
+    ShowElement.className = " m-4 ";
+    let showNameElement = document.createElement("h3");
+    showNameElement.id = show.id;
+    showNameElement.innerHTML = show.name;
+    showNameElement.className = "display-6 text-left p-2 m-3";
+    let showDescriptionElement = document.createElement("div");
+    showDescriptionElement.className = "row";
+    ShowElement.append(showNameElement, showDescriptionElement);
+    let showImageParentElement = document.createElement("div");
+    showImageParentElement.className = "col-md-3 col-12";
+    let showImageElement = document.createElement("img");
+    showImageElement.className =
+      "img-responsive img-thumbnail shadow-lg ml-2 mb-3";
+    if (show.image != null) {
```

Comment:
hi somayeh you code look very good and organised. about question you had on line 111 you put correct condition you said if image is not null but you also need else statment that says no image available  for example if (show.image != null) {

}else{
image = some link
}

Reply:
Hi Ali,
Thank you for your review.
I understand what you mean but I don't know when the image is null , how can I find a link to put on the else or I don't have an idea when it is null.